### ISSUE TITLE
Implement `bulk_load` method that preserves vertex order.

### ISSUE THREAD

**Issue by Seldom-SE** [Stoeoef/spade#77]
I'm getting an error for code that I believe should work. Here's my minimal reproduction:
```Rust
use spade::{ConstrainedDelaunayTriangulation, Point2, Triangulation};

fn main() {
    let vertices = vec![
        Point2::new(60.0, 20.0),
        Point2::new(84.0, 20.0),
        Point2::new(236.0, 20.0),
        Point2::new(60.0, 36.0),
        Point2::new(84.0, 36.0),
        Point2::new(92.0, 28.0),
        Point2::new(60.0, 44.0),
        Point2::new(92.0, 44.0),
        Point2::new(212.0, 44.0),
        Point2::new(20.0, 60.0),
        Point2::new(156.0, 68.0),
    ];

    let edges = vec![(1, 2), (0, 3)];

    let mut triangulation =
        ConstrainedDelaunayTriangulation::<Point2<f32>>::bulk_load(vertices.clone()).unwrap();
    let vertex_handles = triangulation.fixed_vertices().collect::<Vec<_>>();

    for (from, to) in edges {
        println!(
            "{:?} ({:?}), {:?} ({:?})",
            vertices[from], vertex_handles[from], vertices[to], vertex_handles[to]
        );
        triangulation.add_constraint(vertex_handles[from], vertex_handles[to]);
    }
}
```
```
Point2 { x: 84.0, y: 20.0 } (FixedHandle { index: 1 }), Point2 { x: 236.0, y: 20.0 } (FixedHandle { index: 2 })
Point2 { x: 60.0, y: 20.0 } (FixedHandle { index: 0 }), Point2 { x: 60.0, y: 36.0 } (FixedHandle { index: 3 })
thread 'main' panicked at 'Error - constraint edges must not intersect each other', C:\Users\Hayde\.cargo\registry\src\github.com-1ecc6299db9ec823\spade-2.0.0\src\cdt.rs:363:17 
```
I've removed as many points and edges as I can while preserving the error. Note that the edges don't actually intersect:
![image](https://user-images.githubusercontent.com/38388947/191418121-db06cc4e-edf9-4372-b0f0-b98e2de1d5c1.png)
Thanks!

**Comment by Seldom-SE** [comment]
For anyone else running into this issue, my current workaround is unfortunately just to switch to [cdt](https://crates.io/crates/cdt)

**Comment by Stoeoef** [comment]
I think the major issue with attempting to maintain the vertex order is that it doesn't cope well with duplicates. The triangulation can't have two vertices at the same position with different indices - it will have to pick one from the input. Also, indices must be continuous to allow iteration - we can't simply leave gaps in the vertex vector. That being said, I think the feature could be implemented for vertex inputs that don't overlap - something like
`fn bulk_load_stable(vertices: Vec<T>) -> Result<Self, StableBulkLoadError>` which would return an error if a duplicate has been found.

Implementation wise, this should be doable - the existing `fn insert_with_hint_option_impl` (and the method it calls) would need to be extended with an additional `FixedVertexHandle` that expresses that the vertex has already been created.

**Comment by Stoeoef** [comment]
Fixed with version 2.6 (just released)
